### PR TITLE
Add project image build support to the buildPipeline

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/cc @csrwng 
This is super-WIP but getting somewhere

### CODE REVIEW

**Comment by csrwng** [comment]
I like it. /lgtm

File: vars/launchBuildFromRepo.groovy [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/groovy
+
+def call(ctx, jobName, jobId, buildName, fromTag, toTag, sourceImage, sourcePath) {
```
Comment (csrwng) [inline vars/launchBuildFromRepo.groovy]: as mentioned on bjns, jobName, jobId, buildName, sourceImage can be obtained from the ctx
inputs should be fromTag, toTag, dockerfilePath

File: vars/serveRPMs.groovy [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      "--", "/bin/bash", "-c", "'cd /srv/repo && python -m SimpleHTTPServer 8080'"
+    ).out
+    output = new groovy.json.JsonSlurperClassic().parseText(output)
+    for(item in output.items) {
+      if(item.kind == "DeploymentConfig") {
+        item.spec.template.spec.containers[0]["livenessProbe"] = genericProbe()
+        item.spec.template.spec.containers[0]["readinessProbe"] = genericProbe()
+      }
+    }
+    ctx.openshift.apply(groovy.json.JsonOutput.prettyPrint(groovy.json.JsonOutput.toJson(output)))
```
Comment (csrwng) [inline vars/serveRPMs.groovy]: In this case the prettyPrint'ed deploymentConfig won't be logged anywhere, we should log it explicitly

Comment (stevekuznetsov) [reply vars/serveRPMs.groovy]: do we care that it's not? it will get dumped by the artifact gather right

**Comment by stevekuznetsov** [comment]
@csrwng can you PTAL https://github.com/openshift/release-library/pull/33/commits/71c7af51259b732779c6cc1e74beee7cc92c0c29

File: vars/buildPipeline.groovy [hunk 3]
```diff
@@ -51,6 +51,24 @@ def call(String jobName, CloneStep cloneStep, java.util.ArrayList<BuildStep> bui
           }
         }
       }
+      stage("Build Project Images") {
```
Comment (csrwng) [inline vars/buildPipeline.groovy]: This should likely have a "when" clause that only runs if there are any imageBuildSteps 
Also, should create a configmap that will contains the rpm repo URL

**Comment by csrwng** [comment]
What's the JOB_SPEC parameter for?

File: vars/buildPipeline.groovy (hunk 3 above)
Comment (stevekuznetsov) [reply vars/buildPipeline.groovy]: // TODO :)

Comment (stevekuznetsov) [reply vars/buildPipeline.groovy]: still working

File: vars/serveRPMs.groovy (hunk 2 above)
Comment (csrwng) [reply vars/serveRPMs.groovy]: I guess we don't, you're right

**Comment by stevekuznetsov** [comment]
This is "ready"

**Pull request merged**